src/senaite/core/browser/globals/temporary_context.py

```diff
@@ -48,20 +48,31 @@ def __call__(self):
 
         # we expect the portal_type on the first position
         portal_type = self.traverse_subpath[0]
+        context = self.create_temporary_context(portal_type)
+        # call the context to return the rendered html
+        return context()
+
+    def create_temporary_context(self, portal_type):
         portal_types = api.get_tool("portal_types")
         if portal_type not in portal_types:
             raise TypeError("'%s' is not a valid portal_type" % portal_type)
         fti = portal_types[portal_type]
+        tmp_id = "%s_temporary" % portal_type
         if fti.product:
-            raise TypeError("Only Dexterity Types supported")
-        factory = getUtility(IFactory, fti.factory)
-        context = factory("%s_temporary" % portal_type)
+            factory = fti._getFactoryMethod(self.context, check_security=0)
```

Comment:
Since this view supports AT types now too, edit the class comment in accordance

Reply:
Good catch